### ISSUE TITLE
Missing Classmap - Class 'scssc' not found

### ISSUE THREAD

**Issue by Rolice** [leafo/scssphp#375]
Hello,
First of all, I would like to thank you about the great plugin!

Last week we have updated all the packages of our projects one of which is this scssphp. The previously known `classmap.php` file is now gone and all the places using `scssc` class now crash with fatal error:

```php
PHP Fatal error:  Class 'scssc' not found
```

I have not seen any new remapping of the classmap in the `composer.json` file and I think the fix of @robocoder breaks the package:  fixes #361 - **remove deprecated classmap.php file**.

The main problem is that the `scssphp-compass` requires dev-master, and dev-master is now borken. We are in deadlock. I think there are **a lot of packages** which are not working correctly anymore, because of this drastic change.

Do you have any idea why the file is missing and how to fix this issue without modifying the current code, i.e. let the `new scssc` to remain?

Thanks

**Comment by dleffler** [comment]
Have you seen #361?  They moved to using 'namespaces' quite a while back (you can see the comparable namespace equivalent in an older version of classmap.php)

**Comment by Rolice** [comment]
Yes I have seen it, but it breaks a lot of packages this way and we are bound to them. I think there should be some transitional solution, until the packages using this one are ready for the migration.

**Comment by dleffler** [comment]
You had to expect it to be removed at some point since it was deprecated back in v0.1.0 and it finally being pulled in v0.4.0 (which is only in development at this point)...just pointing this out since we moved to namespace calling to get ahead of the curve.  FWIW, you also need to be aware of no longer supporting php v5.4?

**Comment by Rolice** [comment]
@dleffler I am OK, but when another plugin of leafo is expecting dev-master to operate I am in deadlock and I cannot stay on 0.3.3, for example. There are other packages used which also are depended on these classmaps, example is a package of the same author - scssphp-compass, which in addition expects exactly scssphp dev-master. Do I have to fork them all and wait their contributors to accept the changes?

I doubt this would be a solution for any enterprise project.

My code is PHP 5.6 compatible, and looking forward to PHP 7.0.

**Comment by robocoder** [comment]
TL;DR This has been reverted in `master`.

Long story:
* In general, packages shouldn't depend on the `master` branch since `scssphp` is still in active development, and there are breaking changes. We use semantic versioning when tagging releases, so packages should choose the appropriate version constraints to support.
* `classmap.php` **is** the transitional solution and has been marked deprecated for over a year (which I think was ample time for other packages to respond to). The discussion/voting period to finally remove it has elapsed.

**Comment by Rolice** [comment]
@robocoder, agree that master is development branch, but it is deadlock for me, since another package of @leafo, depends on it - [scssphp-compass](https://github.com/leafo/scssphp-compasshttps://github.com/leafo/scssphp-compass). I have applied fork there for that purpose.

**Comment by robocoder** [comment]
`scssphp-compass` is no longer actively maintained/developed. It bundles an old version of compass and only (partially) implements one PHP version of the ruby functions (`compact`)

* http://compass-style.org/index/functions/

**Comment by Rolice** [comment]
Ah, I see. I am not directly involved in the use of these packages and I am not well informed.
Thank you.
